packages/database/src/core/RepoManager.ts

```diff
@@ -87,6 +87,25 @@ export class RepoManager {
     }
   }
 
+  /**
+   * Update an existing repo in place to point to a new host/port.
+   */
+  applyEmulatorSettings(repo: Repo, host: string, port: number): void {
+    repo.repoInfo_ = new RepoInfo(
+      `${host}:${port}`,
```

Comment:
FWIW, I think we need to pick up on our discussion on Friday and figure out whether the "public facing" database URLs (refFromUrl/toString) should use the emulator URLs (as they will with this PR) or if they should expose the production database URLs, which will allow the SDK to work with existing data that hardcodes these URLs.

Reply:
Just pushed a change to address `refFromUrl()`  it's a bit strange at first but I think it's correct:

- `refFromUrl()` must always take an argument that looks like a production URL, not an emulated URL
- If `refFromUrl()` is called after `useEmulator` you get an emulated ref, we just use the path
- If `ref.toString()` is called on a ref that points to the emulator it shows the emulated host, even if it was created with `refFromUrl`

Unless your app relies on `toString()`, this makes emulation a drop-in change.

Reply:
After talking to @yuchenshi, let's update this behavior a bit:

- `toString()` returns `localhost` for the Emulator (same as today)
- `refFromUrl`  only validates if hosts ends in `.firebaseio.com`.

I think that means going back to commits and adding a single if-check. Then we merge :)

Reply:
@schmidt-sebastian ok that's done in my latest commit, PTAL